src/charm.py

```diff
@@ -139,37 +152,24 @@ def __init__(self, *args):
         self.framework.observe(self.on.config_changed, self.set_pod_spec)
         self.framework.observe(self.on["grpc"].relation_changed, self.set_pod_spec)
 
-        self.framework.observe(self.on["grpc-web"].relation_changed, self.send_info)
-
-    def send_info(self, event):
-        if self.interfaces["grpc-web"]:
-            self.interfaces["grpc-web"].send_data(
-                {
-                    "service-host": self.model.app.name,
-                    "service-port": self.model.config["http-port"],
-                }
-            )
```

Comment:
I know we're on the mono-repo bandwagon... but wouldn't this at least make sense to leave separate?  There's no case where another application subscribing to this relation requires a pod_spec change is there?

Reply:
The issue is how `self.interfaces` gets initialized here. It should look something like:

```python
try:
    interfaces = get_interfaces(self)
except Exception as err:
    self.model.unit.status = BlockedStatus(str(err))
    return
```

At which point, you're back to the issue of two handlers fighting over `self.model.unit.status`. I don't really like the mono-handler pattern either, but until https://github.com/canonical/operator/issues/665 gets resolved, it's what we've got :crying_cat_face:

Reply:
Agree with @knkski here, even though it is tempting it will only get us to the same situation again. Lets stick with one eventhandler until we have a real alternative.